src/Umbraco.Core/Publishing/ScheduledPublisher.cs

```diff
@@ -54,7 +59,11 @@ public int CheckPendingAndProcess()
                     throw;
                 }
             }
-            foreach (var d in _contentService.GetContentForExpiration())
+
+            var contentForExpiration = _contentService.GetContentForExpiration().ToArray();
+            if (contentForExpiration.Any())
```

Comment:
Best to not use `Any()` when you know you have a list or array and use the native check like `contentForExpiration.Length > 0` ... this is in fact faster.